src/Rocketlet.ts

```diff
@@ -64,6 +65,17 @@ export abstract class Rocketlet {
     }
 
     /**
+     * This method defines in which relationship to Rocket.Chat this Rocketlet lives.
+     * This information is consumed within the visualization (e. g. for each associated entity,
+     * a different set of visualization options is applicable) and in the persistence.
+     *
+     * @return The entity referred to and the cardinality of this association (e. g. can a room
+     * extended by this Rocketlet have one or multiple Rocketlet data items?
+     *
+     */
+    public abstract getRocketChatAssociation(): IRocketChatAssociation;
```

Comment:
@graywolf336 
You didn't misunderstand the purpose at all. 
Motivation for this arose since I believe the abilities of a Rocketlet should be limited with respect to how they access persistence. Particularly, it should be guided when data can be modified and when it can only be read.
This requires an accessor to be injected which has been created with "proper" abilities by the framework. If you use a factory, you'll depend on the consumer to use it properly which is the second best choice, imho.

Just created #15 to document your objection and to discuss options